.circleci/config.yml

```
@@ -192,13 +192,107 @@ jobs:
+          command: |
+            K8S_DOMAIN=$(hostname -I | awk '{print $1}')
+            export K8S_DOMAIN
+            oc cluster up --server-loglevel=5 --public-hostname=${K8S_DOMAIN}
+            oc login https://${K8S_DOMAIN}:8443 -u developer -p developer --insecure-skip-tls-verify=true
+            # Set environment variables for the CI
+            echo "export K8S_DOMAIN=${K8S_DOMAIN}" >> $BASH_ENV
+            echo "export ARNOLD_DEFAULT_VAULT_PASSWORD=arnold" >> $BASH_ENV
+            echo 'export ANSIBLE_VAULT_PASSWORD="${ARNOLD_DEFAULT_VAULT_PASSWORD}"' >> $BASH_ENV
+            echo "export K8S_AUTH_VERIFY_SSL=no" >> $BASH_ENV
+            echo "export INTERNAL_DOCKER_REGISTRY=172.30.1.1:5000" >> $BASH_ENV
+            source $BASH_ENV
+
+      - run:
+          name: Install arnold CLI
+          command: |
+            wget https://raw.githubusercontent.com/openfun/arnold/master/bin/arnold -O "${HOME}/bin/arnold"
+            chmod +x "${HOME}/bin/arnold"
+
+      - run:
+          name: Setup a new Arnold project
+          command: |
+            arnold -c ralph -e ci setup
+            arnold -d -c ralph -e ci -a ralph create_app_vaults
+            arnold -d -c ralph -e ci -- vault -a ralph decrypt
+            sed -i 's/^# RALPH_/RALPH_/g' group_vars/customer/ralph/ci/secrets/ralph.vault.yml
+            arnold -d -c ralph -e ci -- vault -a ralph encrypt
+            VARS_FILE="group_vars/customer/ralph/ci/main.yml"
+            echo "skip_verification: True" > ${VARS_FILE}
+            echo "apps:" >> ${VARS_FILE}
+            echo "  - name: ralph" >> ${VARS_FILE}
+            echo "ralph_image_name: ${INTERNAL_DOCKER_REGISTRY}/ci-ralph/ralph" >> ${VARS_FILE}
+            echo "ralph_image_tag: ${CIRCLE_SHA1}" >> ${VARS_FILE}
+            echo "ralph_app_replicas: 1" >> ${VARS_FILE}
+            echo "ralph_cronjobs:" >> ${VARS_FILE}
+            echo "  - name: ci-test" >> ${VARS_FILE}
+            echo "    schedule: '* * * * *'" >> ${VARS_FILE}
+            echo "    command: ['date']" >> ${VARS_FILE}
+
+      - run:
+          name: Push ralph image to the k8s cluster docker registry
+          command: |
+            docker login -u developer -p "$(oc whoami -t)" "${INTERNAL_DOCKER_REGISTRY}"
+            docker tag ralph:${CIRCLE_SHA1} "${INTERNAL_DOCKER_REGISTRY}/ci-ralph/ralph:${CIRCLE_SHA1}"
+            arnold -d -c ralph -e ci create_project
+            docker push "${INTERNAL_DOCKER_REGISTRY}/ci-ralph/ralph:${CIRCLE_SHA1}"
+
+      - run:
+          name: Bootstrap ralph application
+          command: |
+            arnold -d -c ralph -e ci -a ralph init
+            arnold -d -c ralph -e ci -a ralph deploy
+            oc project ci-ralph
+            oc get pods -l app=ralph | grep Running
+            test $(oc get cj -o name | wc -l) -eq 1
+            oc create job ralph-ci-test-job --from=cronjob/ralph-ci-test
+            test $(oc get jobs -o name | wc -l) -ge 1
```

What is your intention here ?
Testing that the job has been completed successfully ?

---

At least that it has been created and ran.

---

> At least that it has been created 

Maybe I'm wrong, but I consider that from the moment the `cronjob` creation has been accepted, the underlying `job` can inevitably be created since its definition has been checked.

> and ran

I'm afraid that just after the job creation, the underlying pod will in most case appear as `Pending`.
And even if the job ran, it can be in `Error` status and the test will also pass. This line does not ensure that the job has been ran.

You can wait for the job to be terminated ( [see here](https://stackoverflow.com/a/55077916) ) but it might be overkill :man_shrugging: 


Regardless of these remarks, I have 2 suggestions to be sure to target the right cronjob / jobs : 
```suggestion
            test $(oc get jobs -o name ralph-ci-test-job | wc -l) -ge 1
```

And for the line above : 

```diff
-test $(oc get cj -o name | wc -l) -eq 1
+test $(oc get cj -o name ralph-ci-test | wc -l) -eq 1
```

---

> Maybe I'm wrong, but I consider that from the moment the cronjob creation has been accepted, the underlying job can inevitably be created since its definition has been checked.

I will trust you on this.

> I'm afraid that just after the job creation, the underlying pod will in most case appear as Pending.
And even if the job ran, it can be in Error status and the test will also pass. This line does not ensure that the job has been ran.

True. What would you recommend then? Wait until finished even if it might be overkill?

---

If it was me, I would remove the last 2 lines since we don't check any cronjob execution in any arnold app ;)

But if you want to test it, yes I think the way to go is to wait until the job condition is `complete` or `failed`.

---

Ok, I've tried and it's not that easy, as the wait flag seems to wait forever. Let's be pragmatic on this. :wink: